### PR TITLE
make compatible with queue_classic 3.1.0 RC1

### PR DESCRIPTION
**Pull request by jipiboily**
This PR makes the matchers compatible with the new built-in scheduling in `queue_classic` 3.1. This is only compatible with RC1 as `>= 3.1' didn't seem to work as I was expecting/hoping for with a RC.

This is required by `queue_classic_plus` for its 3.1 compatibility.

I created a `queue_classic_3-0-compatible` branch out of current master.

### CODE REVIEW

File: lib/queue_classic_matchers.rb [hunk 1]
```diff
@@ -6,8 +6,8 @@
 module QueueClassicMatchers
   # Your code goes here...
   module QueueClassicMatchers::QueueClassicRspec
-    def self.find_by_args(queue_name, method, args, table = "queue_classic_jobs")
-      q = "SELECT * FROM #{table} WHERE q_name = $1 AND method = $2 AND args::text = $3"
+    def self.find_by_args(queue_name, method, args)
```
Comment (smathieu) [inline lib/queue_classic_matchers.rb]: This isn't backward compatible, so make sure you make this a minor release, not just a patch release.

Comment (smathieu) [inline lib/queue_classic_matchers.rb]: Nevermind, you did that already :)

File: queue_classic_matchers.gemspec [hunk 2]
```diff
@@ -18,6 +18,7 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
 
+  spec.add_dependency "queue_classic", "3.1.0.RC1"
```
Comment (smathieu) [inline queue_classic_matchers.gemspec]: Will need to be >= before shipping

Comment (jipiboily) [inline queue_classic_matchers.gemspec]: Totally. Unless we ship a RC1 for all the tooling.

**Comment by smathieu** [comment]
:ship:

**Pull request merged**